server/platform/shared/filestore/s3store.go

```diff
@@ -571,18 +618,35 @@ func (b *S3FileBackend) RemoveDirectory(path string) error {
 	return nil
 }
 
-func (b *S3FileBackend) prefixedPath(s string) string {
+func (b *S3FileBackend) prefixedPath(s string) (string, error) {
 	if b.isCloud {
-		// If the request is routed via bifrost, then we need to encode the path
-		// to avoid signature validation errors.
-		// This happens because in bifrost, we are signing the URL outside the SDK
-		// and therefore the signature sent from the bifrost client
-		// will contain the encoded path, whereas the original path is sent
-		// un-encoded.
-		// More info at: https://github.com/aws/aws-sdk-go/blob/a57c4d92784a43b716645a57b6fa5fb94fb6e419/aws/signer/v4/v4.go#L8
-		s = s3utils.EncodePath(s)
-	}
-	return filepath.Join(b.pathPrefix, s)
+		// We do a lookup of the original path for compatibility purposes.
+		exists, err := b._fileExists(filepath.Join(b.pathPrefix, s))
+		if err != nil {
+			var s3Err s3.ErrorResponse
+			if errors.As(err, &s3Err) {
+				// Sometimes S3 will not allow to access other paths.
+				if s3Err.Code != "AccessDenied" {
```

Comment:
To confirm, we can rely on `AccessDenied` to consistently report what amounts to a 404, and not some other error in other situations?

Reply:
They are different. What we are doing is folding "AccessDenied" in the same category as "Not Found". As per my testing, I saw that if you upload images ending with `+` without bifrost, and then use bifrost, it leads to access denied error. We _could_ remove this if you want, because in prod there won't be a situation without bifrost now. But it's just for safety.

Reply:
Yep, as long as there's not a "real" access denied error for which we shouldn't even be trying a different path, I'm good.